src/components/questionnaire/questionnaire-interaction-screen.tsx

```diff
@@ -0,0 +1,160 @@
+import { KeyboardReturn } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  Typography
+} from "@mui/material";
+import { useState } from "react";
+import type { AnswerOption, Question, Questionnaire } from "src/generated/homeLambdasClient";
+import strings from "src/localization/strings";
+
+interface QuestionnaireInteractionScreenProps {
+  questionnaire: Questionnaire | null;
+  mode: "view" | "edit" | "preview";
+  onBack: () => void;
+}
+
+/**
+ *  Screen for the user to interact with the questionnaire; fill, edit and preview
+ */
+const QuestionnaireInteractionScreen: React.FC<QuestionnaireInteractionScreenProps> = ({
+  questionnaire,
+  mode,
+  onBack
+}) => {
+  if (!questionnaire) {
+    return <Typography variant="h4">Loading...</Typography>;
+  }
+
+  const [responses, setResponses] = useState<Record<string, string[]>>({});
```

Comment:
I'm suspicious about this. It's not clear to me what response refers to? Is it users responses while filling out questionnaire?
Also I wonder if this typing might benefit from it's own interface to bring some clarity to the types e.g. 
questionId: string,
selectedOptions: string[]